src/libvirtApi/domain.js

```diff
@@ -1112,3 +1113,21 @@ export function domainUpdateDiskAttributes({ connectionName, objPath, target, re
                 return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainDefineXML', [updatedXML], { timeout, type: 's' });
             });
 }
+
+export async function domainReplaceSpice({ connectionName, id: objPath }) {
+    /* Ideally this would be done by virt-xml, but it doesn't offer that functionality yet
+     * see https://issues.redhat.com/browse/RHEL-17436 */
+    const [domXML] = await call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' });
+    const updatedXML = replaceSpice(domXML);
+
+    // check that updatedXML is valid; if not, throw; it needs to be updated manually
+    await (cockpit.spawn(["virt-xml-validate", "-"], { err: "message" }).input(updatedXML));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1387-20240122-073835-c9864bc3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)